### PR TITLE
configurable topology refresh duration

### PR DESCRIPTION
**Pull request by rishabharyal**
This pull requests a configurable way to topology refresh duration. Two files session.rs and cluster.rs have been modified to set the session configuration and consume the duration configuration respectively.

Note: I am not sure if we need any test cases in this case as the duration is still provided from where its called. 

- [x] I have split my patch into logically separate commits.
- [x] All commit messages clearly explain what they change and why.
- [x] I added relevant tests for new features and bug fixes.
- [x] All commits compile, pass static checks and pass test.
- [x] PR description sums up the changes and reasons why they should be introduced.
- [x] I have provided docstrings for the public items that I want to introduce.
- [x] I have adjusted the documentation in `./docs/source/`.
- [x] I added appropriate `Fixes:` annotations to PR description.

Fixes: #232

### CODE REVIEW

File: docs/source/connecting/connecting.md [hunk 1]
```diff
@@ -34,6 +35,11 @@ async fn main() -> Result<(), Box<dyn Error>> {
 After successfully connecting to some specified node the driver will fetch topology information about
 other nodes in this cluster and connect to them as well.
 
+You can also set the `cluster_topology_refresh_interval` to a non-zero value to periodically refresh the
+cluster topology. This is useful when you do not have unexpected amount of traffic or when you
+have an extra traffic causing topology to change frequently. The default value is 60,
+which means that the driver will refresh the topology every 60 seconds.
+
```
Comment (piodul) [inline docs/source/connecting/connecting.md]: IMO, the way the first sentence is worded ("You can also set the `cluster_topology_refresh_interval` to a non-zero value to periodically refresh the cluster topology") implicitly suggests that the driver does not refresh the cluster topology if you don't set this parameter. This is clarified at the end of the paragraph, but somebody that just skims over the docmentation and reads the beginnings of the paragraphs might miss it.

Personally, I'd structure it like this:

- First mention that the driver periodically refreshes cluster topology,
- Then mention that the interval is 60s by default, but can be changed via `cluster_topology_refresh_interval`.

**Review by piodul — CHANGES_REQUESTED** [review]
To be precise, the parameter that you introduced affects how frequently the whole _cluster metadata_ is refreshed. Cluster metadata not only contains information about the cluster topology, but also information about the cluster schema. The issue #232 was created long ago, when the only metadata that the driver used to fetch in `ClusterWorker::perform_refresh()` was indeed cluster topology, but it has changed since.

I think the current behavior of the parameter is fine and it can control fetching of the whole metadata, but the names used in the code and in the documentation should be adjusted.

Apart from that and some other small comments - looks good.

**Comment by piodul** [comment]
> Note: I am not sure if we need any test cases in this case as the duration is still provided from where its called.

I don't have a good idea on how to write a good test for this. Usually, a `SessionBuilder` test + a manual test to see whether the parameter is properly propagated is sufficient.

**Comment by rishabharyal** [comment]
Hi @piodul, are any further changes needed? Let me know in case you missed this.

**Review by wprzytula — CHANGES_REQUESTED** [review]
There are a number of places with names still containing "topology refresh" instead of "metadata refresh". Please fix them.

File: docs/source/connecting/connecting.md [hunk 2]
```diff
@@ -36,7 +36,7 @@ After successfully connecting to some specified node the driver will fetch topol
 other nodes in this cluster and connect to them as well.
 
 The driver refreshes the cluster metadata periodically, which contains information about cluster topology as well as the cluster schema. By default, the driver refreshes the cluster metadata every 60 seconds. 
-However, you can set the `cluster_topology_refresh_interval` to a non-negative value to periodically refresh the cluster topology. This is useful when you do not have unexpected amount of traffic or when you have an extra traffic causing topology to change frequently.
+However, you can set the `cluster_metadata_refresh_interval` to a non-negative value to periodically refresh the cluster topology. This is useful when you do not have unexpected amount of traffic or when you have an extra traffic causing topology to change frequently.
```
Comment (wprzytula) [inline docs/source/connecting/connecting.md:39]: Still two uses of `topology` where `metadata` would be more appropriate.

**Review by wprzytula — CHANGES_REQUESTED** [review]

File: docs/source/connecting/connecting.md (hunk 2 above)
Comment (rishabharyal) [reply docs/source/connecting/connecting.md]: Yep, took care of that.

**Review by wprzytula — APPROVED** [review]
Thanks, LGTM.
Whoever maintainer is going to merge this, I highly recommend squash&merge.

**Comment by piodul** [comment]
> Hi @piodul, are any further changes needed? Let me know in case you missed this.

@rishabharyal Sorry, I was out on vacation for the last two weeks.

**Pull request merged**